server/render/index.js

```diff
@@ -73,20 +65,29 @@ export function render( element, key = JSON.stringify( element ) ) {
 
 export function serverRender( req, res ) {
 	const context = req.context;
+	let title, metas = [], links = [];
 
 	if ( context.lang !== config( 'i18n_default_locale_slug' ) ) {
 		context.i18nLocaleScript = '//widgets.wp.com/languages/calypso/' + context.lang + '.js';
 	}
 
-	if ( config.isEnabled( 'server-side-rendering' ) &&
-		context.store &&
-		context.layout &&
-		! context.user ) {
-		context.initialReduxState = pick( context.store.getState(), 'ui', 'themes' );
-		const key = context.renderCacheKey || JSON.stringify( context.layout );
-		Object.assign( context, render( context.layout, key ) );
+	if ( config.isEnabled( 'server-side-rendering' ) && ! context.user ) {
+		if ( context.layout ) {
```

Comment:
We do now, see the [discussion](https://github.com/Automattic/wp-calypso/pull/7600#discussion_r81335085).

tl;dr: With the new logic that populates the SSR'd `title` and `meta`s, we want to use the `document-head` subtree's data also for non-iso sections.